src/main/java/cc/chipchop/dao/UserDao.java

```diff
@@ -0,0 +1,62 @@
+package cc.chipchop.dao;
+
+import cc.chipchop.entity.User;
+import cc.chipchop.mapper.UserRowMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Repository
+public class UserDao {
+    private final JdbcTemplate jdbcTemplate;
+    private final UserRowMapper rowMapper = new UserRowMapper();
+
+    @Autowired
+    public UserDao(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+
+    public List<User> findAll() {
+        var query = "SELECT id, email, password FROM users";
+        return this.jdbcTemplate.query(query, this.rowMapper);
+    }
+
+    public Optional<User> findById(long id) {
+        var query = "SELECT id, email, password FROM users WHERE id=?";
+        try {
+            return Optional.ofNullable(this.jdbcTemplate.queryForObject(query, this.rowMapper, id));
+        } catch (EmptyResultDataAccessException e) {
+            return Optional.empty();
+        }
+    }
+
+    public Optional<User> findByEmail(String email) {
+        var query = "SELECT id, email, password FROM users WHERE email=?";
+        try {
+            return Optional.of(Objects.requireNonNull(this.jdbcTemplate.queryForObject(query, this.rowMapper, email)));
+        } catch (EmptyResultDataAccessException e) {
+            return Optional.empty();
+        }
+    }
+
+    public Object insert(User user) {
```

Comment:
You are instructing the query result to be returned as a long with the parameter `Long.class`
```suggestion
    public Long insert(User user) {
```